AlgorithmFactory/Loader.cs

```diff
@@ -315,7 +309,12 @@ public static List<string> GetExtendedTypeNames(Assembly assembly)
                 }
                 catch (ReflectionTypeLoadException e)
                 {
-                    assemblyTypes = e.Types;
+                    assemblyTypes = e.Types.Where(t => t != null).ToArray();
```

Comment:
Very interesting! That's certainly unexpected behavior, I wonder why it behaves that way. I suppose the null entries are the ones where the type didn't load. Perhaps a count of the null entries for the log would also be beneficial? I wonder if `e.LoaderExceptions` has an item for null type, i.e, for each type load failure.

Also, thank you for providing extra context via the linked stack overflow. I think it's good practice to include such a reference either in the source code or in the commit message as well for posterity.

Reply:
It seems indeed that null values in collection do match the types that could not be loaded. And for each such type there is an exception object in LoaderExceptions  collection. Thanks for a review. I made the minor edits according to comments received. commit with those changes follows  ->

Reply:
I use this to count the number of items could not be loaded:
**var countTypesNotLoaded = e.Types.Count(t => t == null);**
Wondering should this be more efficient (faster):
**var countTypesNotLoaded = e.LoaderExceptions.Length;**
if there are exactly as many LoaderExceptions as the types that could not be loaded ?

Reply:
https://docs.microsoft.com/en-us/dotnet/api/system.reflection.reflectiontypeloadexception.loaderexceptions?view=netframework-4.8 says about ReflectionTypeLoadException.LoaderExceptions Property:  The null values in the Types array of this instance line up with the exceptions in this array. So I think it's better use LoaderExceptions.Length.